src/plugins/wizard/public/application/components/top_nav.tsx

```diff
@@ -25,31 +31,57 @@ export const TopNav = () => {
     },
   } = services;
   const rootState = useTypedSelector((state) => state);
+  const dispatch = useTypedDispatch();
   const hasUnappliedChanges = useTypedSelector(
     (state) => !!state.visualization.activeVisualization?.draftAgg
   );
+  if (hasUnappliedChanges) {
+    dispatch(setHasUnappliedChanges({ hasUnappliedChanges: true }));
+  } else {
+    dispatch(setHasUnappliedChanges({ hasUnappliedChanges: false }));
+  }
+
+  const isEmpty = useTypedSelector(
+    (state) => state.visualization.activeVisualization?.aggConfigParams?.length === 0
+  );
+
+  if (isEmpty) {
+    dispatch(setIsEmpty({ isEmpty: true }));
+  } else {
+    dispatch(setIsEmpty({ isEmpty: false }));
+  }
```

Comment:
if `isEmpty` is just a rehash of `state.visualization.activeVisualization?.aggConfigParams?.length === 0`, why cant we just use that? we should not duplicate state information. There should only be one source of truth.

Take a look at creating selecitrs that can derive data https://redux.js.org/usage/deriving-data-selectors